### PR TITLE
Scripts/Commands: Convert argument parsing of lfg commands to new system

### PR DESCRIPTION
**Pull request by Carbenium**
**Changes proposed:**

-  Scripts/Commands: Convert argument parsing of lfg commands to new system

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

### CODE REVIEW

File: src/server/scripts/Commands/cs_lfg.cpp [hunk 1]
```diff
@@ -64,48 +66,45 @@ class lfg_commandscript : public CommandScript
...
+        if (!player)
+            player = PlayerIdentifier::FromTargetOrSelf(handler);
+        if (!player)
             return false;
 
         Group* groupTarget = nullptr;
 
-        if (playerTarget)
-            groupTarget = playerTarget->GetGroup();
+        if (player->IsConnected())
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_lfg.cpp:94]: `if (Player* target = player->GetConnectedPlayer())`

File: src/server/scripts/Commands/cs_lfg.cpp [hunk 2]
```diff
@@ -115,47 +114,36 @@ class lfg_commandscript : public CommandScript
...
-            int32 tmp = atoi(str);
-            if (tmp > -1)
-                options = tmp;
-        }
-
-        if (options != -1)
+        if (optionsArg)
         {
-            sLFGMgr->SetOptions(options);
+            sLFGMgr->SetOptions(*optionsArg);
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_lfg.cpp:133]: what are the values for this? are they some enum?

File: src/server/scripts/Commands/cs_lfg.cpp (hunk 1 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_lfg.cpp:94]: Does it make sense to introduce a local for that though I'm only interested in the group?

File: src/server/scripts/Commands/cs_lfg.cpp (hunk 2 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_lfg.cpp:133]: It's a bitmask. Same as the `DungeonFinder.OptionsMask` config value.

File: src/server/scripts/Commands/cs_lfg.cpp (hunk 1 above)
Comment (Treeston) [reply src/server/scripts/Commands/cs_lfg.cpp:94]: Well, you use that same value in the next line, so...

**Review by Treeston — APPROVED** [review]

**Pull request merged**